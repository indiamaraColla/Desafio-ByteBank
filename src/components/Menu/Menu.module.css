.item {
  text-align: center;
}

.menu {
  width: 180px;
  border-radius: 8px;
  padding: 24px;
  background-color: #f7f7f7;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  list-style: none;
}

.menu > div {
  padding: 24px;
}

.menu > div:not(:last-child) {
  border-bottom: 1px solid #111111;
}

.link {
  color: #111111;
  font-size: 16px;
  cursor: pointer;
  text-decoration: none;
}

.link:hover {
  color: #47a138;
  font-weight: 600;
}

.linkActive {
  color: #47a138;
  font-weight: 600;
  cursor: pointer;
  text-decoration: none;
}

@media (max-width: 1024px) {
  .menu {
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 100%;
    flex-wrap: wrap;
    padding: 12px;
    height: auto;
  }

  .menu > div {
    padding: 12px;
  }

  .menu > div:not(:last-child) {
    border-bottom: none;
  }
}
