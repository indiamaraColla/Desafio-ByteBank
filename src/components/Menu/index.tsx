import { Link } from 'react-router-dom';
import styles from './Menu.module.css';

const listaMenu = [
  { link: 'Início', href: '/' },
  { link: 'Cartões', href: '/cards' },
  { link: 'Serviços', href: '/services' },
  { link: 'Investimentos', href: '/investments' },
];

interface MenuProps {
  path: string;
}

export default function Menu({ path }: MenuProps) {
  return (
    <nav className={styles.menu}>
      {listaMenu.map((item, indice) => {
        return (
          <div key={item.href} className={styles.item}>
            <Link
              to={item.href}
              className={`${styles.link} ${
                path === item.href && styles.linkActive
              }`}
            >
              {item.link}
            </Link>
            {indice !== listaMenu.length - 1 && (
              <div className={styles.divider} />
            )}
          </div>
        );
      })}
    </nav>
  );
}
