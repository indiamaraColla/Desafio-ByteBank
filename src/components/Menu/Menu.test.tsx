import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Menu from './index';

test('Should render link to homepage', () => {
  render(<Menu path="/" />, { wrapper: BrowserRouter });
  const linkPageHome = screen.getByText('InÃ­cio');
  expect(linkPageHome).toBeInTheDocument();
});

test('should render a list with four links', () => {
  render(<Menu path="/" />, { wrapper: BrowserRouter });
  const linkPageHome = screen.getAllByRole('link');
  expect(linkPageHome).toHaveLength(4);
});

test('should not render Extract link', () => {
  render(<Menu path="/" />, { wrapper: BrowserRouter });
  const linkExtract = screen.queryByText('Extrato');
  expect(linkExtract).not.toBeInTheDocument();
});

test('should render the links with the link class', () => {
  render(<Menu path="/" />, { wrapper: BrowserRouter });
  const links = screen.getAllByRole('link');
  links.forEach((link) => {
    expect(link).toHaveClass('link');
  });
  expect(links).toMatchSnapshot();
});
