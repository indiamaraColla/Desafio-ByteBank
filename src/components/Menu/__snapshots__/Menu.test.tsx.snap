// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the links with the link class 1`] = `
Array [
  <a
    class="link linkActive"
    href="/"
  >
    Início
  </a>,
  <a
    class="link false"
    href="/cards"
  >
    Cartões
  </a>,
  <a
    class="link false"
    href="/services"
  >
    Serviços
  </a>,
  <a
    class="link false"
    href="/investments"
  >
    Investimentos
  </a>,
]
`;
